tests/unit/moduleapi/internalsecret.tcl

```diff
@@ -0,0 +1,275 @@
+        assert {[s -1 role] eq {slave}}
+        wait_for_condition 1000 50 {
+            [s -1 master_link_status] eq {up}
+        } else {
+            fail "Master link status is not up"
+        }
+
+        # Execute internal command in master, that will set `x` to `5`.
+        assert_equal {OK} [$master internalauth.internal_rmcall_replicated set x 5]
+
+        # Wait for replica to have the key
+        $slave readonly
+        wait_for_condition 1000 50 {
+            [$slave exists x] eq "1"
+        } else {
+            fail "Test key was not replicated"
+        }
+
+        # See that the slave has the same value for `x`.
+        assert_equal {5} [$slave get x]
+    }
+}
+
+start_cluster 1 0 [list config_lines $modules] {
+    set master [srv 0 client]
+
+    test {Internal commands are not reported in the monitor output for non-internal connections} {
+        # Execute an internal command
+        set rd [redis_deferring_client]
+        $rd monitor
+        $rd read ; # Discard the OK
+        assert_error {*unknown command*} {r internalauth.internalcommand}
+        # Assert that the monitor output does not contain the internal command
+        r ping
+        assert_match {*ping*} [$rd read]
+        $rd close
+    }
+
+    test {Internal commands are reported in the slowlog} {
+        # Authenticate as an internal connection
+        set reply [r internalauth.getinternalsecret]
+        assert_equal {OK} [r auth "internal connection" $reply]
+
+        # Set up slowlog to log all commands
+        r config set slowlog-log-slower-than 0
+
+        # Execute an internal command
+        r slowlog reset
+        r internalauth.internalcommand
+
+        # The slow-log should contain the internal command
+        set log [r slowlog get 1]
+        assert_match {*internalauth.internalcommand*} $log
+    }
+
+    test {Internal commands are reported in the monitor output for internal connections} {
+        # Execute an internal command
+        set rd [redis_deferring_client]
+        $rd monitor
+        $rd read ; # Discard the OK
+        r internalauth.internalcommand
+        assert_match {*internalauth.internalcommand*} [$rd read]
+        $rd close
+    }
+
+    test {Internal commands are reported in the latency report} {
+        # The latency report should contain the internal command
+        set report [r latency histogram internalauth.internalcommand]
+        assert_match {*internalauth.internalcommand*} $report
+    }
+
+    test {Internal commands are reported in the command stats report} {
+        # Execute an internal command
+        r internalauth.internalcommand
+
+        # The INFO report should contain the internal command
+        set report [r info commandstats]
+        assert_match {*internalauth.internalcommand*} $report
+
+        set report [r info latencystats]
+        assert_match {*internalauth.internalcommand*} $report
```

Comment:
Lets also not do it from internal connection, just to make sure.

Reply:
Sure